src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -162,47 +166,43 @@ private void WriteStatistics(string json)
         }
     }
 
-    private void SingleMessageHandling(CancellationToken token, ConsumeResult<string?, byte[]> msg)
+    private async Task<bool> SingleMessageHandlingAsync(CancellationToken token, ConsumeResult<string?, byte[]> msg)
     {
-        _logger.LogDebug(
-            $"Received message from topic '{msg.Topic}:{msg.Partition}' with offset: '{msg.Offset}[{msg.TopicPartitionOffset}]'");
+        _logger.LogDebug("Received message from topic '{Topic}:{Partition}' with offset: '{Offset}[{TopicPartitionOffset}]'",
+            msg.Topic, msg.Partition, msg.Offset, msg.TopicPartitionOffset);
         var cloudEvent = KafkaMessageToCloudEvent(msg.Message);
 
-        var taskAwaiter = ConsumeCallbackAsync?.Invoke(cloudEvent, token).GetAwaiter();
-        taskAwaiter?.OnCompleted(() =>
+        return await await ConsumeCallbackAsync?.Invoke(cloudEvent, token)
```

Comment:
nit: `ConsumeCallbackAsync` shouldn't be able to be null here, as we checked this in the startup. You could exchange the `?` for an `!`.